### DISCUSSION TITLE
useFieldArray conditional field

### DISCUSSION THREAD

**Discussion by MartinCerny-awin** [react-hook-form/react-hook-form#2118] [Q&A] [answered]
I am using useFieldArray where one field is dependent on another one. When I am use delete from useFieldArray, the conditional field gets re-rendered twice and there is observable flickering.

https://codesandbox.io/s/react-hook-form-usefieldarray-template-fmydx?file=/src/index.js

**Comment by bluebill1049** [comment, marked as answer]
This got addressed by the latest patch 6.0.2.

watch('test', fields) // you can use fields as default values so the flickering should go away.

**Reply by ApolloEstrella** [reply]
I have an issue react-hook-form. This is regarding useform watch and fields in fieldarray, they are not in sync
the contents of watch is not the same with "fields". I used watch("items").map instead of the fields.map to reliably get the actual values that I set using setValue. My expected behavior should be is the the fields array should reflect the contents of watch("items"). But there is a trade-off in performance re-rendering using watch("items"), when type on dynamic fields there is a delay.

**Reply by ApolloEstrella** [reply]
I also used watch("items", fields). Same behavior, there is a delay when typing.
